src/app.ts

```diff
@@ -85,8 +88,41 @@ client.on('interactionCreate', async interaction => {
         server = await JoinGuild(interaction.guild)
     }
     await server.EnsureHasRole(interaction.member as GuildMember)
-
-    await ProcessCommand(server, interaction)
+    if(interaction.isCommand()) {
+        await ProcessCommand(server, interaction)
+    }
+    if(interaction.isButton()) {
+        //first 5 characters is the type of interaction, remaining is the name of the queue channel
+        const type = interaction.customId.substring(0,5)
+        const queue_name = interaction.customId.substring(5)
+        //console.log(queue_name + ' app.ts')
+        //console.log(interaction.member?.toString())
+        if(!(interaction.member instanceof GuildMember)) {
+            await interaction.reply({content: `Erm. Somethings wrong, this shouldn't happen. I'll inform the humaniod that maintains me`, ephemeral: true})
+            console.error(`Recieved an interaction without a member from user ${interaction.user} on server ${interaction.guild}`)
+            return;
+        }
+        //console.log(interaction.type + "in app.ts")
+            var errstr = undefined
+        if(type === 'joinn') {
+            errstr = await server.EnqueueUser(queue_name, interaction.member, interaction.type)
+            //console.log('enqueue function ended')
+        } else if(type === 'leave') {
+            errstr = await server.RemoveMember(queue_name, interaction.member, interaction.type)
+            //console.log('remove function ended')
+        } else {
+            console.log('Received invalid button interaction')
+        }
+        //interaction.deferUpdate()
+        if(errstr !== undefined)
```

Comment:
This seems like a strange way to handle errors. For JS promises we usually have a `.catch` in the promise chain. Also, I wouldn't post the error publicly to the channel.

Reply:
I'm pretty sure there isn't a way to send a message to only the person who pressed the button, since unlike slash commands, there isn't a message to reply to. When I pressed a button without having modified the functions, it caused the bot to crash when the UserError function was called. So the only available options I can think of is either sending the error message in chat or via dms. As for using `.catch`, I'll see if I can use that instead

Reply:
Right, so I would catch the `UserError`  from within this function and handle it before it makes it to the outer exception handler. There isn't a message to reply to, so if you have to communicate maybe shoot a direct message instead.